compiler/lexer.nim

```diff
@@ -572,18 +580,27 @@ proc getNumber(L: var TLexer, result: var TToken) =
           len = parseBiggestUInt(result.literal, iNumber)
         except ValueError:
           raise newException(OverflowDefect, "number out of range: " & $result.literal)
-        if len != result.literal.len:
-          raise newException(ValueError, "invalid integer: " & $result.literal)
+        checkLen(len)
         result.iNumber = cast[int64](iNumber)
       else:
         var iNumber: int64
         var len: int
+        var ok = false
         try:
           len = parseBiggestInt(result.literal, iNumber)
-        except ValueError:
+          ok = true
+        except ValueError: discard
+        if uint64Compatible and not ok:
+          try:
+            var iNumber2: uint64
+            len = parseBiggestUInt(result.literal, iNumber2)
+            iNumber = cast[int64](iNumber2)
+            ok = true
+            result.tokType = tkUInt64Lit
+          except ValueError: discard
+        if not ok:
           raise newException(OverflowDefect, "number out of range: " & $result.literal)
```

Comment:
Alternative implementation:

```nim
        try:
          try:
            len = parseBiggestInt(result.literal, iNumber)
          except ValueError:
            if uint64Compatible:
              var uiNumber: uint64
              len = parseBiggestUint(result.literal, uiNumber)
              iNumber = cast[int64](uiNumber)
              result.tokType = tkUInt64Lit
            else:
              raise
        except ValueError:
          raise newException(OverflowDefect, "number out of range: " & $result.literal)
```

This way there's no need to introduce an `ok` variable. But of course I'd say this one should be up to taste.

IMO you should change `iNumber2` into `uiNumber` though.